docs/sections/life_of_a_workflow.md

```diff
@@ -0,0 +1,111 @@
+# Life of a Workflow
+
+This section explains what popper does when it executes a workflow. We will break down what popper does behind the scenes when executing the sample workflow described [here](https://github.com/getpopper/popper/blob/master/docs/sections/getting_started.md).
+
+Each step of a workflow has the following stages:
+
+## 1. Look at `uses` attribute and pull/build image
+
+Each step of a workflow must specify the `DockerFile` images it requires with a `uses` line. For example, the first step of our example workflow contains the following line:
+```
+uses: docker://byrnedo/alpine-curl:0.1.8
+```
+These statements may refer to a `Dockerfile` inside the same repository as the workflow, or to a `Dockerfile` inside an external, public repository or container registry.
+
+The example `uses` line above would result in the following output from Popper:
+```
+[download] docker pull byrnedo/alpine-curl:0.1.8
+```
```

Comment:
it would be useful if this can show what command would need to be executed if users were doing this manually on a terminal. So for each subsection we would show the corresponding command(s).
We could show how this would be done in Docker and Singularity, as a way of showing how popper takes care of handling the differences in CLI syntax for each distinct tool.